modules/apps/foundation/vulcan/vulcan-provider/src/main/java/com/liferay/vulcan/provider/pagination/PaginationParamsContextProvider.java

```diff
@@ -0,0 +1,159 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.vulcan.provider.pagination;
+
+import com.liferay.portal.kernel.util.HttpUtil;
+import com.liferay.portal.kernel.util.MapUtil;
+import com.liferay.vulcan.pagination.Page;
+import com.liferay.vulcan.pagination.PaginationParams;
+
+import java.util.Collection;
+import java.util.Map;
+
+import javax.ws.rs.ext.Provider;
+
+import org.apache.cxf.jaxrs.ext.ContextProvider;
+import org.apache.cxf.message.Message;
+
+/**
+ * @author Alejandro Hernández
+ * @author Carlos Sierra Andrés
+ * @author Jorge Ferrer
+ */
+@Provider
+public class PaginationParamsContextProvider
+	implements ContextProvider<PaginationParams> {
```

Comment:
It seems that this code is coupled to CXF, is that right?

This would not work with any other implementation of JAX-RS?